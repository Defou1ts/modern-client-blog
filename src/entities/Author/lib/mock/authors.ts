import type { AuthorWithLocales } from '../../interfaces';

export const authors: AuthorWithLocales[] = [
	{
		id: 1,
		avatarURL:
			'https://images.pexels.com/photos/7562313/pexels-photo-7562313.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Floyd',
				surname: 'Miles',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Флойд',
				surname: 'Майлз',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 2,
		avatarURL:
			'https://images.pexels.com/photos/16424676/pexels-photo-16424676.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Diana',
				surname: 'Russel',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Дианал',
				surname: 'Рассл',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 3,
		avatarURL:
			'https://images.pexels.com/photos/5704849/pexels-photo-5704849.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Jenny',
				surname: 'Wilson',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Дженни',
				surname: 'Уилсон',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 4,
		avatarURL:
			'https://images.pexels.com/photos/6205509/pexels-photo-6205509.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Leslie',
				surname: 'Alexander',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Александр',
				surname: 'Лесли',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 5,
		avatarURL:
			'https://images.pexels.com/photos/14687031/pexels-photo-14687031.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Guy',
				surname: 'Hawkins',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Гай',
				surname: 'Хавкинс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 6,
		avatarURL:
			'https://images.pexels.com/photos/4467687/pexels-photo-4467687.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Eleanor',
				surname: 'Pena',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Элеонора',
				surname: 'Пена',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 7,
		avatarURL:
			'https://images.pexels.com/photos/5262459/pexels-photo-5262459.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Robert',
				surname: 'Fox',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Роберт',
				surname: 'Фокс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 8,
		avatarURL:
			'https://images.pexels.com/photos/6147118/pexels-photo-6147118.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Jacob',
				surname: 'Jones',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Джейкоб',
				surname: 'Джонс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
];
