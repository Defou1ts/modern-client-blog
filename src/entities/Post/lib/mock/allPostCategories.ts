import BusinessIcon from '@app/assets/icons/business.png';
import StartupIcon from '@app/assets/icons/startup.png';
import EconomyIcon from '@app/assets/icons/economy.png';
import TechnologyIcon from '@app/assets/icons/technology.png';

import type { PostCategory } from '@entities/Post/interfaces';

export const allPostCategories: PostCategory[] = [
	{
		category: 'business',
		icon: BusinessIcon,
		title: {
			en: 'Business',
			ru: 'Бизнес',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'startup',
		icon: StartupIcon,
		title: {
			en: 'Startup',
			ru: 'Стартап',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'economy',
		icon: EconomyIcon,
		title: {
			en: 'Economy',
			ru: 'Экономика',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'technology',
		icon: TechnologyIcon,
		title: {
			en: 'Technology',
			ru: 'Технологии',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
];
