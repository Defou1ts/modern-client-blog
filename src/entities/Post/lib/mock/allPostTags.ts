import type { PostTag } from '../../types';

export const allPostTags: PostTag[] = ['business', 'experience', 'screen', 'technology', 'marketing', 'life'];
