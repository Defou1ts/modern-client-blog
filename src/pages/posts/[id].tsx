import { serverSideTranslations } from 'next-i18next/serverSideTranslations';

import type { GetStaticPaths, GetStaticProps } from 'next';

import { authors } from '@entities/Author/lib/mock/authors';
import { MainContainer } from '@app/wrappers/MainContainer';
import { posts } from '@entities/Post/lib/mock/posts';
import { defaultLocale } from '@shared/contants/defaultLocale';
import { PostPageWrapper } from '@app/wrappers/PostPageWrapper';
import { PostAuthorInfo } from '@widgets/PostAuthorInfo';
import { PostImageOverview } from '@widgets/PostImageOverview';
import { useLocale } from '@shared/hooks/useLocale';
import { JoinOurTeam } from '@widgets/JoinOurTeam';
import { PostContent } from '@entities/Post/ui/PostContent/Index';
import { WhatToReadNext } from '@widgets/WhatToReadNext';

import type { Post } from '@entities/Post/interfaces';
import type { AuthorWithLocales } from '@entities/Author/interfaces';

interface PostPageProps {
	post: Post;
	author: AuthorWithLocales;
	similarPosts: [Post, Post, Post];
	similarPostsAuthors: [AuthorWithLocales, AuthorWithLocales, AuthorWithLocales];
}

const PostPage = ({ author, post, similarPosts, similarPostsAuthors }: PostPageProps) => {
	const { locale } = useLocale();

	const { title, content } = post;

	const translatedTitle = title[locale];
	const translatedContent = content[locale];

	return (
		<MainContainer title={`${translatedTitle} | Modsen client blog`} description={`${translatedTitle}`}>
			<PostPageWrapper>
				<PostAuthorInfo author={author} post={post} />
				<PostImageOverview post={post} />
				<PostContent content={translatedContent} />
				<WhatToReadNext posts={similarPosts} authors={similarPostsAuthors} />
				<JoinOurTeam />
			</PostPageWrapper>
		</MainContainer>
	);
};

export default PostPage;

interface AuthorPagePath {
	params: {
		id: string;
	};
	locale: string;
}

export const getStaticPaths: GetStaticPaths = ({ locales }) => {
	if (locales === undefined) {
		return {
			paths: [],
			fallback: true,
		};
	}

	let paths: AuthorPagePath[] = [];

	locales.forEach((locale) => {
		const pathsWithLocale: AuthorPagePath[] = posts.map((post) => ({
			params: {
				id: post.id.toString(),
			},
			locale,
		}));

		paths = paths.concat(pathsWithLocale);
	});

	return {
		paths,
		fallback: false,
	};
};

export const getStaticProps: GetStaticProps = async ({ params, locale }) => {
	const id = params?.id;

	const post = posts.filter((post) => post.id.toString() === id)[0];
	const author = authors.filter((author) => author.id === post.authorId)[0];
	const similarPosts = posts
		.filter((postItem) => postItem.category === post.category && postItem.id !== post.id)
		.slice(0, 3);
	const similarPostsAuthors = similarPosts.map((post) => authors.filter((author) => author.id === post.authorId)[0]);

	return {
		props: {
			...(await serverSideTranslations(locale ?? defaultLocale, ['common'])),
			post,
			author,
			similarPosts,
			similarPostsAuthors,
		},
	};
};
