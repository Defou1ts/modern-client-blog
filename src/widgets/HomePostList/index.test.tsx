import React from 'react';

import { render, screen } from '@testing-library/react';
import { useRouter } from 'next/router';

import renderer from 'react-test-renderer';

import { HomePostList } from './index';

import { posts } from '@entities/Post/lib/mock/posts';
import { authors } from '@entities/Author/lib/mock/authors';

import type { NextRouter } from 'next/router';
import type { AuthorWithLocales } from '@entities/Author/interfaces';
import type { Post } from '@entities/Post/interfaces';

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}));

describe('HomePostList', () => {
	beforeAll(() => {
		const useRouterMock = {
			locale: 'en',
			locales: ['en', 'ru'],
			pathname: '/',
		};
		(useRouter as jest.Mock<NextRouter>).mockReturnValue(useRouterMock as NextRouter);
	});

	it('renders a  HomePostList', () => {
		render(
			<HomePostList
				featuredPost={posts[0]}
				featuredPostAuthor={authors[0]}
				posts={posts.slice(0, 4) as [Post, Post, Post, Post]}
				postsAuthors={
					authors.slice(0, 4) as [AuthorWithLocales, AuthorWithLocales, AuthorWithLocales, AuthorWithLocales]
				}
			/>
		);
		const text = screen.getByText('home.postlist.featured-post-title');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('HomePostList snapshot', () => {
		const component = renderer.create(
			<HomePostList
				featuredPost={posts[0]}
				featuredPostAuthor={authors[0]}
				posts={posts.slice(0, 4) as [Post, Post, Post, Post]}
				postsAuthors={
					authors.slice(0, 4) as [AuthorWithLocales, AuthorWithLocales, AuthorWithLocales, AuthorWithLocales]
				}
			/>
		);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
