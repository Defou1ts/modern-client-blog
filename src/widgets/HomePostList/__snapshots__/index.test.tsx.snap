// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePostList HomePostList snapshot 1`] = `
<article
  className="wrapper"
  data-test-id="home-post-list"
>
  <div
    className="item"
  >
    <h2
      className="h h2 title"
    >
      home.postlist.featured-post-title
    </h2>
    <div
      className="wrapper"
    >
      <div
        className="imageWrapper"
      >
        <img
          alt="Step-by-step guide to choosing great font pairs"
          data-nimg="1"
          decoding="async"
          height={350}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=https%3A%2F%2Fimages.pexels.com%2Fphotos%2F3184405%2Fpexels-photo-3184405.jpeg%3Fauto%3Dcompress%26cs%3Dtinysrgb%26w%3D1260%26h%3D750%26dpr%3D1&w=1920&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fimages.pexels.com%2Fphotos%2F3184405%2Fpexels-photo-3184405.jpeg%3Fauto%3Dcompress%26cs%3Dtinysrgb%26w%3D1260%26h%3D750%26dpr%3D1&w=750&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.pexels.com%2Fphotos%2F3184405%2Fpexels-photo-3184405.jpeg%3Fauto%3Dcompress%26cs%3Dtinysrgb%26w%3D1260%26h%3D750%26dpr%3D1&w=1920&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          title="Step-by-step guide to choosing great font pairs"
          width={670}
        />
      </div>
      <p
        className="p medium by"
      >
        by
         
        <a
          className="purple"
          href="/author/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Floyd Miles
        </a>
          
        | 
          
        April 25, 2023
      </p>
      <h3
        className="h h3 title"
      >
        Step-by-step guide to choosing great font pairs
      </h3>
      <p
        className="p medium description"
      >
        Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
      </p>
      <a
        className="button"
        data-test-id="home-post-list-featured-link"
        href="/posts/1"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <button
          className="button primary"
        >
          read-more
        </button>
      </a>
    </div>
  </div>
  <div
    className="item"
  >
    <div
      className="postlistTitleWrapper"
    >
      <h2
        className="h h2 title"
      >
        home.postlist.all-posts-title
      </h2>
      <a
        className="viewAll"
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <p
          className="p medium"
        >
          home.postlist.view-all
        </p>
      </a>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="card"
      >
        <p
          className="p medium"
        >
          by
           
          <a
            className="purple"
            href="/author/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Floyd Miles
          </a>
            
          | 
            
          April 25, 2023
        </p>
        <a
          className="link"
          data-test-id="home-postlist-post-link"
          href="/posts/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <h4
            className="h h4"
          >
            Step-by-step guide to choosing great font pairs
          </h4>
        </a>
      </div>
      <div
        className="card"
      >
        <p
          className="p medium"
        >
          by
           
          <a
            className="purple"
            href="/author/2"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Diana Russel
          </a>
            
          | 
            
          April 25, 2023
        </p>
        <a
          className="link"
          data-test-id="home-postlist-post-link"
          href="/posts/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <h4
            className="h h4"
          >
            Design tips for designers that cover everything you need
          </h4>
        </a>
      </div>
      <div
        className="card"
      >
        <p
          className="p medium"
        >
          by
           
          <a
            className="purple"
            href="/author/3"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Jenny Wilson
          </a>
            
          | 
            
          April 25, 2023
        </p>
        <a
          className="link"
          data-test-id="home-postlist-post-link"
          href="/posts/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <h4
            className="h h4"
          >
            How to build rapport with your web design clients
          </h4>
        </a>
      </div>
      <div
        className="card"
      >
        <p
          className="p medium"
        >
          by
           
          <a
            className="purple"
            href="/author/4"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Leslie Alexander
          </a>
            
          | 
            
          April 25, 2023
        </p>
        <a
          className="link"
          data-test-id="home-postlist-post-link"
          href="/posts/4"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <h4
            className="h h4"
          >
            Logo design trends to avoid in 2022
          </h4>
        </a>
      </div>
    </div>
  </div>
</article>
`;
